Clarify documentation for TLS encryption setup (#55739)
elastic/kibana: docs/setup/settings.asciidoc

`elasticsearch.sniffOnConnectionFault:`:: *Default: false* Update the list of
Elasticsearch nodes immediately following a connection fault.

`elasticsearch.ssl.alwaysPresentCertificate:`:: *Default: false* Controls {kib}'s behavior in regard to presenting a client certificate when
requested by {es}. This setting applies to all outbound SSL/TLS connections to {es}, including requests that are proxied for end users.
+

> WARNING: Setting this to `true` when {es} is using certificates to authenticate end users with a PKI realm can lead to proxied requests for

Reviewer: Attempting to simplify this warning:
```suggestion
WARNING:  If {es} uses certificates to authenticate end users with a PKI realm and `elasticsearch.ssl.alwaysPresentCertificate` is `true`, proxied requests for
```

> WARNING: Setting this to `true` when {es} is using certificates to authenticate end users with a PKI realm can lead to proxied requests for

Author: Rewritten with @lcawl's suggestion above!

> end users being executed as the identity tied to the {kib} server.

Reviewer: Continued attempt at simplification:
```suggestion
are executed as the identity that is tied to the {kib} server.
```

Author: This isn't entirely accurate. If this setting is enabled, end users _may_ be mistaken for the Kibana server. I'll apply this suggestion, but change `are` to `may be`.

`elasticsearch.ssl.certificate:` and `elasticsearch.ssl.key:`:: Paths to a PEM-encoded X.509 client certificate and its corresponding
private key. These are used by {kib} to authenticate itself when making outbound SSL/TLS connections to {es}. For this setting to take
effect, the `xpack.security.http.ssl.client_authentication` setting in {es} must be also be set to `"required"` or `"optional"` to request a
client certificate from {kib}.
+
NOTE: These settings cannot be used in conjunction with `elasticsearch.ssl.keystore.path`.

`elasticsearch.ssl.certificateAuthorities:`:: Paths to one or more PEM-encoded X.509 certificate authority (CA) certificates which make up a
trusted certificate chain for {es}. This chain is used by {kib} to establish trust when making outbound SSL/TLS connections to {es}.
+

> In addition to this setting, trusted certificates may be specified via `elasticsearch.ssl.keystore.path` and/or

Reviewer: > In addition to this setting, ....

This seems to imply that you can use all three settings at the same time. Is that correct? If not, I'd recommend "Alternatively, you can specify trusted certificates in ..."

Author: Yes, you can use all three at the same time if you wanted. It's a unlikely scenario but it is possible that you may have some CA certificates in PEM format and some other ones in a PKCS#12 trust store. It was a lot simpler logic to combine than all than to try to throw an error if CA certs were detected in multiple places.

`elasticsearch.ssl.truststore.path`.

`elasticsearch.ssl.keyPassphrase:`:: The password that will be used to decrypt the private key that is specified via
`elasticsearch.ssl.key`. This value is optional, as the key may not be encrypted.

`elasticsearch.ssl.keystore.path:`:: Path to a PKCS#12 keystore that contains an X.509 client certificate and its corresponding private key.
These are used by {kib} to authenticate when making outbound SSL/TLS connections to {es}. For this setting to take effect, the
`xpack.security.http.ssl.client_authentication` setting in {es} must also be set to `"required"` or `"optional"` to request a client
certificate from {kib}.
+
--
If the keystore contains any additional certificates, those will be used as a trusted certificate chain for {es}. This chain is used by
{kib} to establish trust when making outbound SSL/TLS connections to {es}. In addition to this setting, trusted certificates may be

> specified via `elasticsearch.ssl.certificateAuthorities` and/or `elasticsearch.ssl.truststore.path`.

Author: Yes, as CA certificates are read from the filesystem from all three settings, they are all combined and used together as trusted certificates. Order doesn't matter (thank God), it's just a big list of CA certificates that gets looped over during verification.

NOTE: This setting cannot be used in conjunction with `elasticsearch.ssl.certificate` or `elasticsearch.ssl.key`.
--

`elasticsearch.ssl.keystore.password:`:: The password that will be used to decrypt the keystore that is specified via
`elasticsearch.ssl.keystore.path`. If the keystore has no password, leave this unset. If the keystore has an empty password, set this to
`""`.

`elasticsearch.ssl.truststore.path:`:: Path to a PKCS#12 trust store that contains one or more X.509 certificate authority (CA) certificates
which make up a trusted certificate chain for {es}. This chain is used by {kib} to establish trust when making outbound SSL/TLS connections
to {es}.
+

> In addition to this setting, trusted certificates may be specified via `elasticsearch.ssl.certificateAuthorities` and/or

Reviewer: Per earlier comment, it would be nice to clarify why you'd use one or the other of these truststore/keystore/CA settings.

Author: Copy-pasting my other response:
Yes, you can use all three at the same time if you wanted. It's a unlikely scenario but it is possible that you may have some CA certificates in PEM format and some other ones in a PKCS#12 trust store. It was a lot simpler logic to combine than all than to try to throw an error if CA certs were detected in multiple places.

`server.socketTimeout:`:: *Default: "120000"* The number of milliseconds to wait before closing an
inactive socket.

`server.ssl.certificate:` and `server.ssl.key:`:: Paths to a PEM-encoded X.509 server certificate and its corresponding private key. These
are used by {kib} to establish trust when receiving inbound SSL/TLS connections from end users.
+
NOTE: These settings cannot be used in conjunction with `server.ssl.keystore.path`.

`server.ssl.certificateAuthorities:`:: Paths to one or more PEM-encoded X.509 certificate authority (CA) certificates which make up a
trusted certificate chain for {kib}. This chain is used by {kib} to establish trust when receiving inbound SSL/TLS connections from end

> users. This chain is also used by {kib} to verify client certificates from end users when PKI authentication is enabled.

Reviewer: ```suggestion
users. If PKI authentication is enabled, this chain is also used by {kib} to verify client certificates from end users.
```
